src/main/java/com/example/hbs/service/UserService.java

```diff
@@ -50,20 +52,21 @@ public UserResponseDto addUser(UserRequestDto userRequestDto) {
     }
 
     public UserResponseDto updateUser(Long id, UserRequestDto userRequestDto) {
-        if (userRepository.findById(id).isPresent()) {
-            User user = userRepository.findById(id).get();
-            if (userRequestDto.getUserName() != null)
-                user.setUserName(userRequestDto.getUserName());
-            if (userRequestDto.getUserEmail() != null)
-                user.setUserEmail(userRequestDto.getUserEmail());
-            if (userRequestDto.getUserContact() != null)
-                user.setUserContact(userRequestDto.getUserContact());
-            if (userRequestDto.getUserRole() != null)
-                user.setUserRole(userRequestDto.getUserRole());
-            return userMapper.map(userRepository.save(user));
-        } else {
-            return null;
+        User user = userRepository.findById(id).orElse(null);
+        Optional.ofNullable(user).orElseThrow(() -> new HbsException("No such user Id found"));
+        if (userRequestDto.getUserName() != null) {
+            user.setUserName(userRequestDto.getUserName());
         }
+        if (userRequestDto.getUserEmail() != null) {
+            user.setUserEmail(userRequestDto.getUserEmail());
+        }
+        if (userRequestDto.getUserContact() != null) {
+            user.setUserContact(userRequestDto.getUserContact());
+        }
+        if (userRequestDto.getUserRole() != null) {
+            user.setUserRole(userRequestDto.getUserRole());
+        }
+        return userMapper.map(userRepository.save(user));
```

Comment:
we should throw an exception if some required value is missing
like role, email etc.